src/main/java/com/elytradev/concrete/resgen/ConcreteResourcePack.java

```diff
@@ -0,0 +1,328 @@
+		simpleItemModel = simpleItemModel.replaceAll("#L0", textureLocation);
+		cache.put(name, simpleItemModel);
+		return simpleItemModel;
+	}
+
+	private Item getItem(String name) {
+		String itemID = name.substring(name.lastIndexOf("/") + 1, name.lastIndexOf("."));
+		if (Item.getByNameOrId(modID + ":" + itemID) != null) {
+			return Item.getByNameOrId(modID + ":" + itemID);
+		} else {
+			ResourceLocation location = nameToLocation(name);
+			try {
+				HashBiMap<Pair<RegistryDelegate<Item>, Integer>, ModelResourceLocation> customModelsMap = HashBiMap.create(getCustomModels());
+				String resourcePath = location.getResourcePath();
+				resourcePath = resourcePath.substring(resourcePath.lastIndexOf("/") + 1);
+				resourcePath = resourcePath.substring(0, resourcePath.lastIndexOf("."));
+				String domain = location.getResourceDomain();
+				location = new ModelResourceLocation(domain + ":" + resourcePath, isLocation(name, "/models/item/") ? "inventory" : "normal");
+				if (customModelsMap.inverse().containsKey(location))
+					return customModelsMap.inverse().get(location).getLeft().get();
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+			return Items.AIR;
+		}
+	}
+
+	/**
+	 * Attempt to get metadata value from ModelLoader to account for custom mesh locations, returns 0 if none were found.
+	 *
+	 * @param name name provided by getInputStreamByName
+	 * @return metadata if found, 0 if none.
+	 */
+	private Integer getMetaFromName(String name) {
+		ResourceLocation location = nameToLocation(name);
+		try {
+			HashBiMap<Pair<RegistryDelegate<Item>, Integer>, ModelResourceLocation> customModelsMap = HashBiMap.create(getCustomModels());
+			String resourcePath = location.getResourcePath();
+			resourcePath = resourcePath.substring(resourcePath.lastIndexOf("/") + 1);
+			resourcePath = resourcePath.substring(0, resourcePath.lastIndexOf("."));
+			String domain = location.getResourceDomain();
+			location = new ModelResourceLocation(domain + ":" + resourcePath, isLocation(name, "/models/item/") ? "inventory" : "normal");
+			if (customModelsMap.inverse().containsKey(location))
+				return customModelsMap.inverse().get(location).getRight();
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		return 0;
+	}
+
+	@Override
+	protected boolean hasResourceName(String name) {
+		if ((boolean) hasResourceName.invoke(realResourcePack, name))
+			return false;
+
+		if (isLocation(name, "/blockstates/")) {
+			LOG.debug("Location was provided " + name + ", matched blockstate check.");
+			return true;
+		}
+		if (isLocation(name, "/models/block/")) {
+			LOG.debug("Location was provided " + name + ", matched block model check.");
+			return true;
+		}
+		if (isLocation(name, "/models/item/")) {
+			LOG.debug("Location was provided " + name + ", matched item model check.");
+			return true;
+		}
+		return false;
+	}
+
+	@Override
+	public Set<String> getResourceDomains() {
+		return realResourcePack.getResourceDomains();
```

Comment:
Since this is no longer a wrapper, shouldn't it return `Collections.singletonSet(modid)`?